lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -0,0 +1,256 @@
+module ActiveMerchant
+  module Billing
+    class PayeezyGateway < Gateway
+      class_attribute :integration_url
+
+      self.test_url = 'https://api-cert.payeezy.com/v1/transactions'
+      self.integration_url = 'https://api-cat.payeezy.com/v1/transactions'
+      self.live_url = 'https://api.payeezy.com/v1/transactions'
+
+      self.default_currency = 'USD'
+      self.money_format = :cents
+      self.supported_countries = %w(US CA)
+
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover, :jcb, :diners_club]
+
+      self.homepage_url = 'https://developer.payeezy.com/'
+      self.display_name = 'Payeezy'
+
+      CREDIT_CARD_BRAND = {
+        'visa' => 'Visa',
+        'master' => 'Mastercard',
+        'american_express' => 'American Express',
+        'discover' => 'Discover',
+        'jcb' => 'JCB',
+        'diners_club' => 'Diners Club'
+      }
+
+      def initialize(options = {})
+        requires!(options, :apikey, :apisecret, :token)
+        super
+      end
+
+      def purchase(money, creditcard, options = {})
+        commit(:purchase, purchase_or_auth_request(money, creditcard, options))
+      end
+
+      def authorize(money, creditcard, options = {})
+        commit(:authorize, purchase_or_auth_request(money, creditcard, options))
+      end
+
+      def capture(money, authorization, options = {})
+        commit(:capture, capture_or_credit_request(money, authorization, options))
+      end
+
+      def credit(money, authorization, options = {})
```

Comment:
This should be `refund` not `credit`. `credit` does an unreferenced push of funds to a card.